src/python/WMCore/ReqMgr/Web/ReqMgrService.py

```diff
@@ -521,10 +524,13 @@ def fetch(self, rid):
                 transitions.remove(status)
             visible_attrs = get_modifiable_properties(status)
             filterout_attrs = get_protected_properties()
-
             # extend filterout list with "RequestStatus" since it is passed separately
             filterout_attrs.append("RequestStatus")
 
+            for key, val in self.assignArgs.items():
+                if key not in doc or not doc.get(key):
```

Comment:
Alan, I'm not 100% sure if "not doc.get(key)" will work, the problem is what do you want from it. If empty string or empty list are acceptable arguments, then the conditions will not give you what you want and it will overwrite doc[key] value with defaults. I think a proper way should be:
```
if not doc.get(key, None)
```
which does both tests, i.e. check that key in dict and if not it returns None which is not a proper data-type for assignment args. Compare this to empty string or empty list which may be proper values (and data-types) for assigned args.